src/core/expr/fexpr_list.cc

```diff
@@ -323,7 +323,7 @@ void FExpr_List::prepare_by(
         outflags.push_back(SortFlag::DESCENDING);
       } else {
         outwf.cbind( arg->evaluate_n(ctx) );
-        outflags.push_back(SortFlag::NONE);
+        outflags.push_back(ctx.get_sort_direction());
```

Comment:
I guess this logic may have a problem. For example, if both `negcol` and `ctx.get_sort_direction()` are `true`, the order will be descending, though it should be ascending:

```python
>>> from datatable import dt, f
>>> DT = dt.Frame(range(5))
>>> DT[:, :, sort(-f[0])]
   | C0
-- + --
 0 |  4
 1 |  3
 2 |  2
 3 |  1
 4 |  0

[5 rows x 1 column]
>>> DT[:, :, sort(-f[0], reverse=True)] # nothing changes, but it should
   | C0
-- + --
 0 |  4
 1 |  3
 2 |  2
 3 |  1
 4 |  0

[5 rows x 1 column]
```

Reply:
so if we have double negatives (`negcol` and `reverse=True`), the result should be positive (that is ascending), is that how it should be? is that how it's done in R-datatable as well?

Reply:
Samuel has raised a good point (see below). Having double negatives does not seem like a good design. I would rather go with either deprecating the reverse or the behaviour to be descending (like in your example above) despite double negatives.

Reply:
Yes, double negative normally results in a positive, see below

```python
>>> DT[:, :, sort(-(-f[0]))]
   | C0
-- + --
 0 |  0
 1 |  1
 2 |  2
 3 |  3
 4 |  4

[5 rows x 1 column]
```

Why it should be descending? Minus sign just specifies what value should be used for sorting, in the above case it is sorted by `-f[0]`, and the `reverse` parameter controls the overall sorting direction: whether it should be done asc or desc. So from the user point of view the result of `DT[:, :, sort(-f[0], reverse=True)]` should be asc sorting.

With respect to deprecating the `reverse` parameter, I would say it can be pretty useful in some cases. For instance, sometimes one doesn't know in advance if the sort should be done asc or desc, it may be an input boolean parameter or whatever else, then it is easier and more readable to write something like `DT[:, :, sort(f[0], reverse=MY_INPUT_PARAMETER)` than converting `MY_INPUT_PARAMETER` to +-1 and then multiply it by `f[0]`. 

May be @st-pasha also had some other use-cases in mind when [proposed](https://github.com/h2oai/datatable/issues/792) the API.

Reply:
Btw, when I use the column index, instead of the `f[]`,  the `reverse` parameter still doesn't work

```python
>>> from datatable import dt, f
>>> DT = dt.Frame(range(5))
>>> DT[:, :, dt.sort(0, reverse=True)]
   | C0
-- + --
 0 |  0
 1 |  1
 2 |  2
 3 |  3
 4 |  4

[5 rows x 1 column]

>>> DT[:, :, dt.sort(f[0], reverse=True)]
   | C0
-- + --
 0 |  4
 1 |  3
 2 |  2
 3 |  1
 4 |  0

[5 rows x 1 column]
```

Reply:
`Btw, when I use the column index, instead of the f[], the reverse parameter still doesn't work`

Good observation. This doesn't work on master either and I think (correct me if I am wrong), this is not directly related to how we deal with the reverse flag. Shall we create a new issue for this or combine it in the same issue? I vote for the former.